src/main/java/de/srendi/advancedperipherals/common/util/inventory/FluidUtil.java

```diff
@@ -30,6 +30,41 @@ public class FluidUtil {
     private FluidUtil() {
     }
 
+    public static int moveFluid(IFluidHandler inventoryFrom, IFluidHandler inventoryTo, FluidFilter filter) {
+        if (inventoryFrom == null) return 0;
+
+        int amount = filter.getCount();
+        int transferableAmount = 0;
+
+        // The logic changes with storage systems since these systems do not have slots
+        if (inventoryFrom instanceof IStorageSystemFluidHandler storageSystemHandler) {
+            FluidStack extracted = storageSystemHandler.drain(filter, IFluidHandler.FluidAction.SIMULATE);
+            int inserted = inventoryTo.fill(extracted, IFluidHandler.FluidAction.EXECUTE);
+
+            transferableAmount += storageSystemHandler.drain(filter.setCount(inserted), IFluidHandler.FluidAction.EXECUTE).getAmount();
+
+            return transferableAmount;
+        }
+
+        if (inventoryTo instanceof IStorageSystemFluidHandler storageSystemHandler) {
+            if (filter.test(inventoryFrom.getFluidInTank(0))) {
+                FluidStack toExtract = inventoryFrom.getFluidInTank(0).copy();
+                toExtract.setAmount(amount);
+                FluidStack extracted = inventoryFrom.drain(toExtract, IFluidHandler.FluidAction.SIMULATE);
+                if (extracted.isEmpty())
+                    return 0;
+                int inserted = storageSystemHandler.fill(extracted, IFluidHandler.FluidAction.EXECUTE);
+
+                extracted.setAmount(inserted);
+                transferableAmount += inventoryFrom.drain(extracted, IFluidHandler.FluidAction.EXECUTE).getAmount();
+            }
+
+            return transferableAmount;
+        }
+
```

Comment:
what if both inventory is not `IStorageSystemFluidHandler`? Is this situation not considered?

Reply:
Nope, that would be for example a RS System to RS System transfer which is not supported/AP does not offer that in any way